mender-update/update_module/v3/platform/boost_filesystem/fs_operations.cpp

```diff
@@ -227,6 +250,159 @@ expected::ExpectedStringVector DiscoverUpdateModules(const conf::MenderConfig &c
 	return ret;
 }
 
+Error UpdateModule::PrepareStreamNextPipe() {
+	download_.stream_next_path_ = path::Join(update_module_workdir_, "stream-next");
+
+	if (::mkfifo(download_.stream_next_path_.c_str(), 0600) != 0) {
+		int err = errno;
+		return error::Error(
+			generic_category().default_error_condition(err),
+			"Unable to create `stream-next` at " + download_.stream_next_path_);
+	}
+	return error::NoError;
+}
+
+Error UpdateModule::OpenStreamNextPipe(ExpectedWriterHandler open_handler) {
+	auto opener = make_shared<AsyncFifoOpener>(download_.event_loop_);
+	download_.stream_next_opener_ = opener;
+	return opener->AsyncOpen(download_.stream_next_path_, open_handler);
+}
+
+Error UpdateModule::PrepareAndOpenStreamPipe(
+	const string &path, ExpectedWriterHandler open_handler) {
+	auto fs_path = fs::path(path);
+	boost::system::error_code ec;
+	if (!fs::create_directories(fs_path.parent_path(), ec) && ec) {
```

Comment:
Curiosity, I just learnt that `cpp` actually has an `and` keyword :exploding_head:

Reply:
Q: Can this ever fail without setting the error code?

Reply:
> Curiosity, I just learnt that `cpp` actually has an `and` keyword exploding_head

I vaguely recall having heard about that, yes! Interestingly, it's not listed as part of any particular standard. Has it always been there..? :open_mouth: 

I'm fine with either, but `&&` is pretty established, so I'm not going on a replacement spree! 😄

> Q: Can this ever fail without setting the error code?

I don't think so. If so there is a bug in Boost. It's the same kind of scenario as with `errno`, I suppose. Library functions are supposed to set them, and we can't really do anything but trust them.